packages/sdk-components-react/src/html-embed-patchers.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const isDOMContentLoaded = () => {
+  return (
+    document.readyState === "complete" || document.readyState === "interactive"
+  );
+};
+
+const eventListenerTasks: Array<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  () => void
+> = [];
+
+let domContentLoadedPatched = false;
+
+export const patchDomEvents = () => {
+  // If original event is not fired yet, do nothing as it can cause serious side effects.
+  if (isDOMContentLoaded() === false) {
+    console.error("DOMContentLoaded event has not been fired yet");
+    return;
+  }
+
+  if (domContentLoadedPatched) {
+    return;
+  }
+
+  domContentLoadedPatched = true;
+
+  console.info("Patching DOMContentLoaded event listener");
+
+  const originalAddEventListener = document.addEventListener;
+  const originalWindowAddEventListener = window.addEventListener;
+
+  const domContentLoadedEvent = new Event("DOMContentLoaded");
+  const windowLoadEvent = new Event("load");
```

Comment:
its probably better not to reuse those events, in case callee decides to mutate it